vmngclient/api/partition_manager_api.py

```diff
@@ -91,26 +91,45 @@ def remove_partition(self, devices: List[Device], version: str, force: bool = Fa
         url = "/dataservice/device/action/removepartition"
         payload = {
             "action": "removepartition",
-            "devices": self.device_versions.get_device_list_in_available(version, devices),
+            "devices": self.device_versions.get_device_list_in_available(version, devices, True),
             "deviceType": "vmanage",
         }
         if force is False:
-            invalid_devices = self._check_remove_partition_possibility(payload["devices"])
+            invalid_devices = self._check_remove_partition_possibility(cast(list, payload["devices"]))
             if invalid_devices:
                 raise ValueError(
                     f"Current or default version of devices with ids {invalid_devices} \
                         are equal to remove version. Action denied!"
                 )
-        remove_action: Dict[str, str] = self.repository.session.post(url, json=payload).json()
+        remove_action: Dict[str, str] = self.session.post(url, json=payload).json()
+        return remove_action["id"]
+
+    def remove_available_partitions(self, devices: List[Device]) -> str:
+        """
+        Remove all available partitions for devices in devices list.
+
+        Args:
+            devices (List[Device]): _description_
+
+        Returns:
+            str: remove action id
+        """
+        url = "/dataservice/device/action/removepartition"
+        payload = {
+            "action": "removepartition",
+            "devices": self.device_versions.get_devices_available_versions(devices),
+            "deviceType": "vmanage",
+        }
+        remove_action: Dict[str, str] = self.session.post(url, json=payload).json()
```

Comment:
also goldening. Following method will do a job 

def get_available_partitions()

then user will use get_available_partitions and remove_partitions

Reply:
Same here, I removed that, and one common method is used to remove partitions.